packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js

```diff
@@ -247,7 +247,6 @@ export default function () {
                     if (id.isIdentifier()) {
                       addTo(exports, id.node.name, id.node);
                       init.replaceWith(buildExportsAssignment(id.node, init.node).expression);
-                      nonHoistedExportNames[id.node.name] = true;
```

Comment:
I assume based on the tests results, that instead of removing this, you need to tighten up the conditions that is it run.